src/SonarScanner.MSBuild.Tasks/GetAnalyzerSettings.cs

```diff
@@ -326,14 +327,17 @@ private AnalyzerSettings GetLanguageSpecificSettings(AnalysisConfig config)
 
         /// <summary>
         /// Merges and returns the supplied list of file paths. In case of duplicate
-        /// // file *names* (not full paths), the path from the primary list is used.
+        /// // file *names* (not full paths) of SonarAnalyzers, the path from the primary list is used.
         /// </summary>
         private string[] MergeFileLists(IEnumerable<string> primaryList, IEnumerable<string> secondaryList)
```

Comment:
initially this method was generic, now it's not generic anymore - could you reflect it in the method name and/or parameter names, variable names?

Because the method expects 
- the `primaryList` to contain SonarAnalyzers
- the `secondaryList` not to contain SonarAnalyzers (if it contains, it removes them)

That was the initial implicit logic as well, if I am not mistaken.

**My feeling is that this method is making some assumptions on what `primaryList` should contain and should come from, and what `secondaryList` might contain and where it should come from, and this assumption is not reflected in the parameter name.**

Reply:
this is used in both:
- CreateLegacyProductProjectSettings
- CreateMergedAnalyzerSettings

the behavior change shouldn't break any of those methods assumptions, right?

Reply:
> this is used in both:
> 
> * CreateLegacyProductProjectSettings
> * CreateMergedAnalyzerSettings
> 
> the behavior change shouldn't break any of those methods assumptions, right?

If my understanding is correct, the `CreateLegacyProductProjectSettings` is when `roslyn.ignoreIssues` is set to `true` meaning only our analyzers are embeded. (This will of course change in the near future by working on https://github.com/SonarSource/sonar-scanner-msbuild/issues/1211 ) However the "correct" behavior is always to only remove our analyzers and do not mess with what the user provided to the build.